### PR TITLE
Ewl 534 match hero block

### PR DESCRIPTION
**Pull request by ra-phael**
https://eurosportdigital.atlassian.net/browse/EWL-534

Ended up splitting into 2 different components as they require different sets of props.

### CODE REVIEW

File: src/modules/MatchHero/index.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+`;
+
+const MatchHero = ({ title, date, hour, labels, cTAText, cTALink, marketingMessage }) => (
+  <StyledWrapper>
+    <Labels labels={labels} />
+    <StyledContent>
+      <StyledBasicInfo>
+        <H1>{title}</H1>
+        <StyledTime>
+          {date} | Starting from {hour}
```
Comment (arturkin) [inline src/modules/MatchHero/index.js]: Why not to use label here?

File: src/modules/MatchHero/index.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  hour: '',
+  marketingMessage: '',
+};
+
+MatchHero.propTypes = {
+  title: PropTypes.string.isRequired,
+  date: PropTypes.string,
+  hour: PropTypes.string,
+  labels: labelsType.isRequired,
+  cTAText: PropTypes.string.isRequired,
```
Comment (arturkin) [inline src/modules/MatchHero/index.js]: Caps is typo or it's intentional?

File: src/modules/MatchHero/index.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+export const MatchHeroWithScore = ({ gameStatus, scoreData, labels, displayWatchButton, onWatchButtonClick }) => (
+  <StyledWrapper>
+    <Labels labels={labels} />
+    <StyledScoreWrapper>
+      <SetsScore data={scoreData} />
+    </StyledScoreWrapper>
+    {displayWatchButton && (
+      <PlayIconLink bgColorIcon={colors.utahCrimson} bgColorText={colors.venetianRed} onClick={onWatchButtonClick}>
+        {gameStatus === 'after' ? 'WATCH REPLAY' : 'WATCH LIVE'}
```
Comment (arturkin) [inline src/modules/MatchHero/index.js]: Why not to use labels passed from props?

File: src/modules/MatchHero/index.js (hunk 1 above)
Comment (ra-phael) [reply src/modules/MatchHero/index.js]: The Labels component?

File: src/modules/MatchHero/index.js (hunk 2 above)
Comment (ra-phael) [reply src/modules/MatchHero/index.js]: Intentional as it's an acronym (Call To Action) but I don't have strong opinions about that. :)

File: src/modules/MatchHero/index.js (hunk 3 above)
Comment (ra-phael) [reply src/modules/MatchHero/index.js]: Good idea, that'll remove some logic from this component.

File: src/modules/MatchHero/index.js (hunk 1 above)
Comment (arturkin) [reply src/modules/MatchHero/index.js]: No I meant string should not be hardcoded

File: src/modules/MatchHero/index.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  min-width: 282px;
+  max-width: 368px;
+  ${breakpoints.medium(css`
+    margin-bottom: 43px;
+  `)};
+`;
+
+const MatchHero = ({ title, date, hour, labels, cTAText, cTALink, marketingMessage }) => (
+  <StyledWrapper>
+    <Labels labels={labels} />
```
Comment (arturkin) [inline src/modules/MatchHero/index.js]: Please add picture component to display hero background-image!

File: src/modules/MatchHero/index.js (hunk 1 above)
Comment (val3ntin) [inline src/modules/MatchHero/index.js]: separator `|` should be styled here...

File: src/modules/MatchHero/index.js (hunk 4 above)
Comment (ra-phael) [reply src/modules/MatchHero/index.js]: After discussing with David Garry we thought it'd be better if the zone is in charge of injecting the player within itself and therefore also handle the display of a background image in the right place.

File: src/modules/MatchHero/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ra-phael) [reply src/modules/MatchHero/index.js]: I removed gameStatus as it wasn't used at all

File: src/modules/MatchHero/index.js (hunk 1 above)
Comment (ra-phael) [reply src/modules/MatchHero/index.js]: done

File: src/modules/MatchHero/index.test.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        scoreData={scoreDataDuring}
+        displayWatchButton
+        onWatchButtonClick={() => null}
+        watchButtonText="WATCH LIVE"
+      />
+    );
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('should display the watch button', () => {
```
Comment (val3ntin) [inline src/modules/MatchHero/index.test.js]: i think this test can be removed as watchButton display already into your snapshot from line 72

File: src/modules/MatchHero/index.test.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+        scoreData={scoreDataDuring}
+        displayWatchButton
+        onWatchButtonClick={() => null}
+      />
+    );
+    const playButton = wrapper.find({ bgColorText: colors.venetianRed });
+    expect(playButton).toHaveLength(1);
+  });
+
+  it('should display a text on watch button passed as props', () => {
```
Comment (val3ntin) [inline src/modules/MatchHero/index.test.js]: the same - tested as a part of your snapshot.

File: src/modules/MatchHero/index.test.js [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+        marketingMessage="Watch Eurosport live"
+      />
+    );
+  });
+
+  it('should render as expected', () => {
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('should render labels', () => {
```
Comment (val3ntin) [inline src/modules/MatchHero/index.test.js]: remove it - it's tested as a part of the snapshot above ( line 29)

File: src/modules/MatchHero/index.test.js [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    expect(wrapper.find(Labels)).toHaveLength(1);
+  });
+
+  it('should render a CTA', () => {
```
Comment (val3ntin) [inline src/modules/MatchHero/index.test.js]: tested as a part of the snapshot above ( line 29)

File: src/modules/MatchHero/index.test.js [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+    expect(wrapper.find(StyledCTA)).toHaveLength(1);
+  });
+
+  it('should render a marketing message', () => {
```
Comment (val3ntin) [inline src/modules/MatchHero/index.test.js]: tested as a part of the snapshot above ( line 29)

File: src/modules/MatchHero/index.test.js [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+    expect(
+      wrapper
+        .find(StyledMarketingMessage)
+        .childAt(0)
+        .text()
+    ).toEqual('Watch Eurosport live');
+  });
+
+  it('should render Basic Info', () => {
```
Comment (val3ntin) [inline src/modules/MatchHero/index.test.js]: tested as a part of the snapshot above ( line 29)

File: src/modules/MatchHero/index.test.js [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+        onWatchButtonClick={() => null}
+        watchButtonText="WATCH REPLAY"
+      />
+    );
+    const playButton = wrapper.find({ bgColorText: colors.venetianRed });
+    expect(playButton).toHaveLength(1);
+    expect(playButton.prop('children')).toEqual('WATCH REPLAY');
+  });
+
+  it('should render SetScore', () => {
```
Comment (val3ntin) [inline src/modules/MatchHero/index.test.js]: covered in your snapshot as well ?

**Comment by arturkin** [comment]
Last thing I noticed - it's a bad text wrap for the tablet and mobile views:
image

**Pull request merged**